doc/api/esm.md

```diff
@@ -175,88 +175,75 @@ unspecified.
 capabilities are limited: it only defines the main entry point of the package.
-The `"exports"` field, part of [Package Exports][], provides an alternative to
-`"main"` where the package main entry point can be defined while also
-encapsulating the package, preventing any other entry points besides those
-defined in `"exports"`. If package entry points are defined in both `"main"` and
-`"exports"`, the latter takes precedence in versions of Node.js that support
-`"exports"`. [Conditional Exports][] can also be used within `"exports"` to
-define different package entry points per environment.
 
-#### `package.json` `"main"`
+The `"exports"` field provides an alternative to `"main"` where the package
+main entry point can be defined while also encapsulating the package, preventing
+any other entry points besides those defined in `"exports"`. If package entry
+points are defined in both `"main"` and `"exports"`, the latter takes precedence
+in versions of Node.js that support `"exports"`. [Conditional Exports][] can
+also be used within `"exports"` to define different package entry points per
+environment.
 
-The `package.json` `"main"` field defines the entry point for a package,
-whether the package is included into CommonJS via `require` or into an ES
-module via `import`.
+#### Package Exports Main
+
+To set the main entry point for a package, it is advisable to use both
+`"exports"` and the `"main"`:
 
 <!-- eslint-skip -->
 ```js
-// ./node_modules/es-module-package/package.json
 {
-  "type": "module",
-  "main": "./src/index.js"
+  "main": "./main.js",
+  "exports": "./main.js"
 }
 ```
 
-```js
-// ./my-app.mjs
-
-import { something } from 'es-module-package';
-// Loads from ./node_modules/es-module-package/src/index.js
-```
-
-An attempt to `require` the above `es-module-package` would attempt to load
-`./node_modules/es-module-package/src/index.js` as CommonJS, which would throw
-an error as Node.js would not be able to parse the `export` statement in
-CommonJS.
-
-As with `import` statements, for ES module usage the value of `"main"` must be
-a full path including extension: `"./index.mjs"`, not `"./index"`.
-
-If the `package.json` `"type"` field is omitted, a `.js` file in `"main"` will
-be interpreted as CommonJS.
+The benefit of doing this is that when using the `"exports"` field all
+subpaths of the package will no longer be available to importers under
+`require('pkg/subpath.js')`, and instead they will get a new error,
+`ERR_PACKAGE_PATH_NOT_EXPORTED`.
 
-The `"main"` field can point to exactly one file, regardless of whether the
-package is referenced via `require` (in a CommonJS context) or `import` (in an
-ES module context).
+This new encapsulation of exports provides new more reliable guarantees
+about package interfaces for tools and when handling semver upgrades for a
+package. It is not a strong encapsulation since
+`require('/path/to/pkg/subpath.js')` will still work correctly.
```

Comment:
I’m confused; you’re saying that the string form is not equivalent to an object with the string under a dot? I thought any use of “exports” always by default restricted subpaths, otherwise what’s the point of using the string form? (the below example seems to suggest my intuition, which makes this sentence seem incorrect)

Reply:
The example is showing the use of an absolute path. Would it be clearer if I explicitly state eg:

```suggestion
a require of the absolute path to any package `require('/path/to/pkg/subpath.js')`
subpath will still work correctly.
```

Reply:
ahh, yes, that would be clearer. altho since that absolute path wouldn't be portable, it doesn't feel like a useful example since nobody would do it in the first place.

(separately, why would an absolute path work here? i'd expect it to work, or not, the same as `require('./node_modules/pkg/subpath.js')`)

Reply:
Ok I've updated the example to include node_modules to maybe make this clearer. Yes it's not a portable pattern - you'd usually want to use require.resolve in such a workflow (eg tools can use this for package.json lookups that break with exports encapsulations, although they really shouldn't be relying on require to get internal package.jsons of other packages).

With regards to the note, relative paths work equally here - it is only the bare specifier entry that gives the not exported error. The reason being that there is always some way around this anyway. We could discuss further locking down in future though if people want to bring this up.